### PR TITLE
Use full test suite on make test

### PR DESCRIPTION
**Pull request by cyphar**
Enable the full test suite to run on `make test`. They also all run
inside a Docker container for maximum reproducibility.

Signed-off-by: Aleksa Sarai <asarai@suse.de>

### CODE REVIEW

**Comment by hqhq** [comment]
I don't think snifftest is needed anymore, it's all covered by integration test, so I tried to remove it in https://github.com/opencontainers/runc/pull/785 WDYT?

**Comment by cyphar** [comment]
I don't really have a strong opinion, but it does appear redundant to have a sniff test that is then replicated in an integration test.

/cc @duglin -- do you have any reason why we should keep it in?

**Comment by cyphar** [comment]
I'll respond in the issue.

**Comment by hqhq** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -35,24 +43,24 @@ man:
...
-	docker create --name=$(RUNC_INSTANCE) $(RUNC_IMAGE)
-	docker cp $(RUNC_INSTANCE):$(RUNC_BUILD_PATH) .
-	docker rm $(RUNC_INSTANCE)
-
 integration: runctestimage
 	docker run -e TESTFLAGS -t --privileged --rm -v $(CURDIR):/go/src/$(PROJECT) $(RUNC_TEST_IMAGE) make localintegration
 
-localintegration:
+localintegration: all
 	bats tests/integration${TESTFLAGS}
```
Comment (mlaventure) [inline Makefile]: Would it be possible to always run this in TAP mode via `-t` while at it? Jenkins doesn't log very nicely the output otherwise.

**Comment by cyphar** [comment]
@mlaventure I've updated it to use `-t` mode.

**Comment by mlaventure** [comment]
@cyphar Thanks, output is much more readable now!

**Comment by cyphar** [comment]
Are you okay with this @mrunalp @crosbymichael ?

**Comment by crosbymichael** [comment]
i got this when running locally after the unit test run

```
PASS
ok      _/go/src/github.com/opencontainers/runc/libcontainer/xattr      0.004s
Makefile:56: recipe for target 'localunittest' failed
make[1]: *** [localunittest] Error 1
make[1]: Leaving directory '/go/src/github.com/opencontainers/runc'
Makefile:50: recipe for target 'localtest' failed
make: *** [localtest] Error 2
Makefile:53: recipe for target 'unittest' failed
make[1]: *** [unittest] Error 2
make[1]: Leaving directory '/home/michael/development/gocode/src/github.com/opencontainers/runc'
Makefile:47: recipe for target 'test' failed
make: *** [test] Error 2
```

**Comment by cyphar** [comment]
@crosbymichael Are you running the tests as root? Also, there doesn't appear to be an error message in your output.

**Comment by crosbymichael** [comment]
@cyphar i thought it was supposed to run them in the container now so i'm just running `make test` as my normal user

**Comment by cyphar** [comment]
Ah, I see now that you were running `make test` (the error trace is hard to read in `make`)? I thought you meant "locally" as in `make localtest` (which requires root). I can't seem to reproduce that locally on this branch.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**